qemu/deps/win_driver_install/win_driver_install.py

```diff
@@ -0,0 +1,171 @@
+def certutil(vol_utils):
+    certutil_cmd = "certutil -addstore -f TrustedPublisher %s\\redhat.cer" % vol_utils
+    if not os.path.exists("C:\Windows\system32\certutil.exe"):
+        shutil.copy("%s\certutil.exe" % vol_utils, "C:\Windows\system32")
+    if not os.path.exists("C:\Windows\system32\certadm.dll"):
+        shutil.copy("%s\certadm.dll" % vol_utils, "C:\Windows\system32")
+    print "Install certificate !"
+    try:
+        subprocess.call(certutil_cmd, shell=True)
+    except Exception, err:
+        error_msg = "Can not install certificate with cmd %s!\n" % certutil_cmd
+        error_msg += "Details refers: %s" % err
+        print error_msg
+        sys.exit(1)
+
+
+# Install the specific driver
+def install_driver(driver_path, driver_name, vol_utils):
+    install_driver_cmd = "C:\dpinst.exe /A /PATH %s /C /LM /Q /F" % driver_path
+    certutil(vol_utils)
+    print "Install driver %s!" % driver_name
+    try:
+        subprocess.call(install_driver_cmd, shell=True)
+    except Exception, err:
+        error_msg = "Driver install failed with cmd %s!\n" % install_driver_cmd
+        error_msg += "Details refers: %s and dpinst.log" % err
+        print error_msg
+        sys.exit(1)
+
+
+def get_inf_files(driver_path, driver_name):
+    inf_name = "%s.inf" % driver_name
+    inf_files = []
+    for root, dirs, files in os.walk(driver_path):
+        files_path = map(lambda x: os.path.join(root, x), files)
+        inf_files += filter(lambda x: x.endswith(inf_name), files_path)
+    return inf_files
+
+
+# Uninstall the specific driver
+def uninstall_driver(driver_name):
+    driver_store = "C:\Windows\System32\DriverStore\FileRepository"
+    uninstall_driver_cmd = "C:\dpinst.exe /U %s /C /LM /Q /D"
+    if not os.path.exists(driver_store):
+        print "Driver store path %s does not exist." % driver_store
+        sys.exit(1)
+    print "Uninstall driver !"
+    inf_files = get_inf_files(driver_store, driver_name)
+    uninstall_driver_command = map(lambda x: uninstall_driver_cmd % x,
+                                   inf_files)
+    try:
+        for cmd in uninstall_driver_command:
+            subprocess.call(cmd, shell=True)
+    except Exception, err:
+        error_msg = "Driver uninstall failed with cmd %s!\n" % uninstall_driver_command
+        error_msg += "Details refers: %s and dpinst.log" % err
+        print error_msg
+        sys.exit(1)
+
+
+# Get the specific current driver version
+def get_current_driver_ver(device_name):
+    key = "\d*\.\d*\.\d*\.\d*"
+    get_driver_ver_cmd = "wmic path win32_pnpsigneddriver where" \
+                         " Devicename='%s' get driverversion" % device_name
+    driver_version = os.popen(get_driver_ver_cmd).read()
+    return re.findall(key, driver_version, re.M)[-1].strip()
+
+
+# Get expected driver version
+def get_expected_driver_ver(driver_path, driver_name):
+    key = "\d*\.\d*\.\d*\.\d*"
+    inf_file = get_inf_files(driver_path, driver_name)

```

Comment:
type  of inf_file is list, so I guess that  below lines will cause OSError Exception.